### PR TITLE
refactor: NoticeBar - marquee.native, close #1800.

### LINKED ISSUE #1800: NoticeBar在iOS上运行正常, 在Android上不滚动显示是静止的
### Version
1.6.5

### Environment
react-native 0.47.1

### Reproduction link
[https://codepen.io/paranoidjk/pen/LWpaKe](https://codepen.io/paranoidjk/pen/LWpaKe)

### Steps to reproduce
<NoticeBar marqueeProps={{loop:true,style:{color:'#666666',fontSize:13}}}  icon={null}>
       Notice: The arrival time of incomes and transfers of Yu &#39;E Bao will be delayed during National Day.
</NoticeBar>

### What is expected?
Android能和iOS一样在字数超过范围时能滚动显示

### What is actually happening?
Android在使用NoticeBar的时候字数超过显示范围时静止的

### PR DESCRIPTION
**Pull request by zhang740**
- 将create-react-class写法改成class
- fix: android下onLayoutContainer晚于onLayout，并且onLayout触发频繁导致的问题
- fix: android下切分Text的方法动画性能不足，改换单Text，扩大包围View的方式

 ref #1815
---
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1816)

### CODE REVIEW

File: components/notice-bar/Marquee.native.tsx [hunk 1]
```diff
@@ -1,113 +1,114 @@
...
 export interface MarqueeProp {
-  text: string;
+  text: React.ReactNode;
   loop?: boolean;
   leading?: number;
   trailing?: number;
   className?: string;
   fps?: number;
+  style?: ViewStyle;
+  maxWidth?: number;
```
Comment (silentcloud) [inline components/notice-bar/Marquee.native.tsx]: 加了这个 maxWidth 感觉不太灵活？

Comment (paranoidjk) [reply components/notice-bar/Marquee.native.tsx]: 默认不是满屏的吗？

@silentcloud 这个你跟进处理下吧？ @zhang740 要搞网站改版

Comment (silentcloud) [reply components/notice-bar/Marquee.native.tsx]: 嗯 我来

Comment (zhang740) [reply components/notice-bar/Marquee.native.tsx]: maxWidth是为了真的特别长需要进行控制的，默认值1000应该足够了，或者调大默认值，如果超出尺寸会显示 'xxx...' 这种形式，这种处理方式会有这个限制，好处是View数少。 @silentcloud

**Comment by silentcloud** [comment]
好，我先合了，剩下的问题我来处理

**Pull request merged**